### ISSUE TITLE
pointcloud to spc trilinear interpolation

### ISSUE THREAD

**Issue by 000rrr** [NVIDIAGameWorks/kaolin#644]
Thanks for making this library available. I have an issue with trilinear interpolation on a spc.

I have a point cloud with rgb features, and I convert the point cloud to an spc. The pointcloud is a volumetric sphere and has a linear gradient in the z direction from rgb(1.0,0.0,0.0) at (x,y,-1.0) to rgb(0.0,1.0,0.0) at (x,y,0.0) to rgb(0.0,0.0,1.0) at (x,y,1.0). 
The point cloud is shown below.

![image](https://user-images.githubusercontent.com/113952695/199634717-02903452-eeab-4ef1-99f2-2464f67105d5.png)

I am trying to trilinear interpolate the rgb values according to given coordinates in this example at (0.0, 0.0, 0.0). Using this code from your example.

```
verts = torch.tensor(pos).view(l,3).cuda()
color = torch.tensor(data).view(l,3).cuda()


spc = kaolin.ops.conversions.pointcloud.unbatched_pointcloud_to_spc(verts,level=4,features=color)

point_hierarchy_dual, pyramid_dual = kaolin.ops.spc.unbatched_make_dual( point_hierarchy=spc.point_hierarchies , pyramid=spc.pyramids[0])
trinkets, _ = kaolin.ops.spc.unbatched_make_trinkets(spc.point_hierarchies, spc.pyramids[0], point_hierarchy_dual, pyramid_dual)

query_coord = torch.tensor((0.0, 0.0, 0.0)).unsqueeze(0).cuda()  
sampled_query_coords = query_coord.unsqueeze(1)


query_index = kaolin.ops.spc.unbatched_query(spc.octrees, spc.exsum, query_coord, spc.max_level, with_parents=False)
interpolated = kaolin.ops.spc.unbatched_interpolate_trilinear(sampled_query_coords, query_index.int(), spc.point_hierarchies, trinkets, spc.features, 4)

print( interpolated )
```

However, I get the result, 

`tensor([[[1., 0., 1.]]], device='cuda:0')`

which should simply be (0.0,1.0,0.0).

Could you kindly let me know where I am mistaken here? Do I have to define the features on the leaf node of the dual octree? If so, how would I transfer them efficiently?

Thanks.

**Comment by tovacinni** [comment]
Thanks for your interest in Kaolin!

You do in fact need to define the features on the leaf node of the dual octree. If you have a function that calculates those colors from positions, you can use the 4th level of `point_hierarchy_dual` (since you are interpolating the 4th level) and use those coordinates to calculate the colors.

**Comment by 000rrr** [comment]
@tovacinni thanks so much for your response. That makes sense. I got it working. Thank you.

![image](https://user-images.githubusercontent.com/113952695/199863855-44224381-bd74-4d45-9929-e34c2b9ca5c9.png)

**Comment by tovacinni** [comment]
Awesome to hear!!!
